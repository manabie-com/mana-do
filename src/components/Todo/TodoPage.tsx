import { DeleteOutlined } from '@ant-design/icons'
import { notification } from 'antd'
import React, { useState, useEffect } from 'react'
import moment from 'moment'
import todosAPI from '../../service/todos'
import { Todos } from '../../models/todos'
import Presenter from './Presenter'
const openNotification = (status: string, content: string) => {
  if (status === 'success') {
    notification.success({
      duration: 3,
      message: 'Success',
      description: content,
    })
  } else {
    notification.error({
      duration: 3,
      message: 'Error',
      description: content,
    })
  }
}
const Todo: React.FC = () => {
  const [radio, setRadio] = useState<string>('default')
  const [todosValue, setTodosValue] = useState<string>('')
  const [todosTemp, setTodosTemp] = useState<Todos[]>()
  const [selectedRowKeys, setSelectedRowKeys] = useState<any>()
  const [loading, setLoading] = useState<boolean>(true)
  const [visible, setVisible] = useState<boolean>(false)
  const [valueRow, setValueRow] = useState<Todos[]>()
  const [todos, setTodos] = useState<Todos[]>()
  const onSelectChange = (selectedRowKeys: Todos, value: Todos[]): void => {
    setSelectedRowKeys(selectedRowKeys)
    setValueRow(value)
  }
  const showDrawer = () => {
    setVisible(true)
  }
  const onClose = () => {
    setVisible(false)
    setValueRow([])
  }
  const rowSelection: any = {
    selectedRowKeys,
    valueRow,
    onChange: onSelectChange,
  }
  const getAllTodos = async () => {
    try {
      setLoading(true)
      const res = await todosAPI.getAll()
      // check status api
      if (res.status === 200) {
        setTodos(res.data)
      }
      setLoading(false)
    } catch (error) {
      setLoading(false)
    }
  }
  useEffect(() => {
    getAllTodos()
  }, [])
  const handleDeleteItem = async (record: Todos) => {
    try {
      if (record && record.id && record.id !== '') {
        const res = await todosAPI.deleteTodos(record.id)
        if (res.status === 200 || res.status === 201) {
          await getAllTodos()
          openNotification(
            'success',
            `Successfully deleted todos "${record.content}"`,
          )
        }
      }
    } catch (error) {}
  }
  const columns = [
    {
      title: 'Content',
      dataIndex: 'content',
      render: (text: any, record: any) => (
        <div
          onClick={() => handleUpdateObject(record)}
          style={{ color: 'blue', cursor: 'pointer' }}
        >
          {text}
        </div>
      ),
    },
    {
      title: 'UserID',
      dataIndex: 'user_id',
    },
    {
      title: 'Created Date',
      dataIndex: 'created_date',
    },
    {
      title: 'Status',
      dataIndex: 'status',
    },
    {
      title: 'Action',
      dataIndex: 'action',
      render: (text: any, record: any) => (
        <DeleteOutlined
          onClick={() => handleDeleteItem(record)}
          style={{ color: 'red', fontSize: '1.5rem', cursor: 'pointer' }}
        />
      ),
    },
  ]
  const handleTodosValue = (e: any): void => {
    const { value } = e.target
    setTodosValue(value)
  }
  const createTodos = async (object: Todos) => {
    try {
      setLoading(true)
      const res = await todosAPI.createTodos(object)
      // check status api
      if (res.status === 200 || res.status === 201) {
        await getAllTodos()
        setTodosValue('')
        openNotification(
          'success',
          `Create todos with content ${object.content} successfully.`,
        )
      }
      setLoading(false)
    } catch (error) {
      setLoading(false)
    }
  }
  const handleComplete = (status: string): void => {
    let flag = 0
    // check todos content
    if (todosValue === '') {
      openNotification('error', 'You have not written content for todos')
    } else {
      // check content exist
      if (todos && todos.length > 0) {
        todos.forEach((values, index) => {
          if (values.content === todosValue) {
            flag++
          }
        })
      }
      // todos doesn't have this conent
      if (flag === 0) {
        // get user_id from localStorage
        const user_id = localStorage.getItem('user_id')
        let now = moment()
        // get date now using moment
        const object = {
          content: todosValue,
          user_id: String(user_id),
          created_date: now.format('YYYY-MM-DD, HH:mm:ss'),
          status: status,
        }
        createTodos(object)
      } else {
        openNotification('error', `Content ${todosValue} already exists`)
      }
    }
  }
  const handleRadio = async (e: any) => {
    const { value } = e.target
    setSelectedRowKeys([])
    if (value === 'default') {
      await getAllTodos()
      setTodosTemp([])
      setRadio(value)
      openNotification('success', 'Show all successful todos.')
    } else {
      setRadio(value)
      // check array todos
      if (todos && todos.length > 0) {
        setLoading(true)
        const result = todos.filter((values, index) => {
          // filter all todos status === radio user
          return values.status.toLowerCase() === value.toLowerCase()
        })
        // check array result
        if (result && result.length > 0) {
          setTodosTemp([...result])
        }
        setLoading(false)
        openNotification(
          'success',
          `Filter todos by status ${value} successful.`,
        )
      }
    }
  }
  const handleAll = async () => {
    await getAllTodos()
    setTodosTemp([])
    setRadio('default')
    setSelectedRowKeys([])
    openNotification('success', 'Show all successful todos.')
  }
  const handleTodosDeleteAll = async (id: string) => {
    try {
      setLoading(true)
      const res = await todosAPI.deleteTodos(id)
      if (res.status === 200 || res.status === 201) {
        await getAllTodos()
        if (todos && todos.length > 0) {
          await handleDeleteAll()
        } else {
          openNotification('success', 'Clear all todos successfully')
        }
      }
      setLoading(false)
    } catch (error) {
      setLoading(false)
    }
  }
  const handleDeleteAll = async () => {
    // check todos before delete all
    if (todos && todos.length > 0) {
      // Run a for loop to call api delete each object
      todos.forEach((values) => {
        if (values && values.id && values.id !== '') {
          handleTodosDeleteAll(values.id)
        }
      })
    }
  }
  const handleUpdateTodos = async (object: Todos) => {
    try {
      setLoading(true)
      const res = await todosAPI.updateTodos(object)
      if (res.status === 200 || res.status === 201) {
        await getAllTodos()
        setSelectedRowKeys([])
        openNotification(
          'success',
          `Update todos "${object.content}" was successful.`,
        )
        onClose()
        setValueRow([])
      }
      setLoading(false)
    } catch (error) {
      setLoading(false)
    }
  }
  const handleUpdate = (): void => {

    // check array
    valueRow &&
      valueRow.forEach((values, index) => {
        handleUpdateTodos({ ...values })
      })
  }
  const handleUpdateObject = (object: Todos): void => {
    const arrayTemp = valueRow && valueRow.length > 0 ? [...valueRow] : []
    arrayTemp.push(object)
    setValueRow([...arrayTemp])
    setVisible(true)
  }
  return (
    <Presenter
      todosValue={todosValue}
      onClose={onClose}
      visible={visible}
      handleUpdate={handleUpdate}
      valueRow={valueRow}
      todos={todos}
      todosTemp={todosTemp}
      columns={columns}
      rowSelection={rowSelection}
      handleRadio={handleRadio}
      radio={radio}
      loading={loading}
      showDrawer={showDrawer}
      selectedRowKeys={selectedRowKeys}
      handleTodosValue={handleTodosValue}
      handleAll={handleAll}
      handleComplete={handleComplete}
      handleDeleteAll={handleDeleteAll}
    />
  )
}
export default Todo
