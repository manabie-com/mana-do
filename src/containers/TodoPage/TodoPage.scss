@import "../../assets/styles/main.scss";

.Todo__container {
  border: 1px solid rgba(0, 0, 0, 0.13);
  border-radius: 0.5rem;
  text-align: left;
  width: 500px;
  padding: 1.5rem;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 0.5rem 1.5rem;
  background-color: #f9faff;
  border-radius: 1rem;
  min-height: 80%;
  display: flex;
  flex-direction: column;
  min-height: 80vh;

  .Todo__list {
    display: flex;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}
