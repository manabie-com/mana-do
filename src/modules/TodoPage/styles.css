button {
  outline: none;
  border: none;
  box-shadow: 2px 0 2px currentColor;
  border-radius: 4px;
  max-height: 25px;
  padding: 4px 8px;
}

button:hover {
  opacity: 0.85;
}

input[type='checkbox'] {
  width: 15px;
  height: 15px;
  box-shadow: none;
  border: none;
  outline: none;
  margin: 0px 0px 0px 10px;
}

input[type='checkbox']:focus {
  box-shadow: none;
  border: none;
  outline: none;
}

input {
  min-height: 36px;
  border: none;
  outline: none;
  padding: 0 12px;
  box-shadow: 2px 0 4px rgba(0, 0, 0, 0.2);
  border-radius: 4px;
}

input:focus {
  box-shadow: 1px 0 9px rgba(0, 0, 0, 0.25);
}

.ToDo__container {
  border: 1px solid rgba(0, 0, 0, 0.13);
  border-radius: 8px;
  width: 500px;
  margin-top: 5rem;
  padding: 0px 24px 24px 24px;
  box-shadow: 2px 2px 1px rgba(0, 0, 0, 0.09), 3px 2px 3px rgba(0, 0, 0, 0.05);
}

.Todo__creation {
  display: flex;
}

.Todo__input {
  flex: 1 1;
}

.ToDo__list {
  display: flex;
  flex-direction: column;
  margin-top: 1.5rem;
}

.Todo__toolbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 24px;
  min-height: 36px;
}

.Todo__tabs {
  display: flex;
  justify-content: center;
}

.Todo__tabs > *:not(:last-child) {
  margin-right: 8px;
}

.Action__btn {
  box-shadow: none;
  border: 1px solid #555555;
  color: #555555;
  background-color: #fff;
}
.Selected__btn {
  color: #fff;
  background-color: #304d95;
}
.delete__all__btn {
  color: #e1000a;
  background-color: #f9d0d9;
}
