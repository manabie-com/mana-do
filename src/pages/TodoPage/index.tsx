import React, { useEffect, useReducer, useRef, useState } from "react";
import "./index.css";

import reducer, { initialState } from "store/reducer";
import {
  setTodos,
  createTodo,
  toggleAllTodos,
  deleteAllTodos,
  updateTodoStatus,
  updateTodoContent,
  deleteTodo,
} from "store/actions";
import Service from "service";
import { TodoStatus } from "models/todo";
import { TodoInput, ToDoToolbar } from "components";

type EnhanceTodoStatus = TodoStatus | "ALL";

export const TodoPage = () => {
  const [{ todos }, dispatch] = useReducer(reducer, initialState);
  const [showing, setShowing] = useState<EnhanceTodoStatus>("ALL");
  const inputRef = useRef<any>(null);

  useEffect(() => {
    (async () => {
      const resp = await Service.getTodos();
      dispatch(setTodos(resp || []));
    })();
  }, []);

  const onCreateTodo = async (content: string) => {
    const resp = await Service.createTodo(content);
    dispatch(createTodo(resp));
  };

  const onUpdateTodoStatus = (status: boolean, todoId: any) => {
    dispatch(updateTodoStatus(todoId, status));
  };

  const onToggleAllTodo = (status: boolean) => {
    dispatch(toggleAllTodos(status));
  };

  const onDeleteTodo = (todoId: string) => {
    dispatch(deleteTodo(todoId));
  };

  const onDeleteAllTodo = () => {
    dispatch(deleteAllTodos());
  };

  const onEditingTodo = (newContent: string, todoId: string) => {
    dispatch(updateTodoContent(todoId, newContent));
  };

  return (
    <div className="ToDo__container disable-select">
      <TodoInput inputRef={inputRef} _onCreateTodo={onCreateTodo} />

      <div className="legend-bar">
        <div className="legend-bar-item todo-color-active"></div>
        <div className="legend-bar-item todo-color-complete"></div>
      </div>

      <ToDoToolbar
        todoLength={todos.length}
        showing={showing}
        onToggleAllTodo={onToggleAllTodo}
        onDeleteAllTodo={onDeleteAllTodo}
        onSetShowing={setShowing}
      />
    </div>
  );
};
