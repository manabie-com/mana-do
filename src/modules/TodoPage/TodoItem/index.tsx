import React, { useState, useRef, useEffect } from 'react';
import "./styles.css"
import { Todo, TodoStatus } from '../../../models/todo';

interface TodoItemProps {
  todo: Todo,
  onChangeStatus: Function,
  onDeleteTodo: Function,
  onUpdateTodoContent: Function
}

const TodoItem = (props: TodoItemProps) => {
  const { todo, onChangeStatus, onDeleteTodo, onUpdateTodoContent } = props
  const [edit, setEdit] = useState(false)
  const spanRef = useRef<any>(null);

  useEffect(() => {
    function handleClickOutside(event: any) {
      if (spanRef.current && !spanRef.current.contains(event.target)) {
        spanRef.current.textContent = todo.content
        setEdit(false)
      }
    }

    if (edit) {
      document.addEventListener("mousedown", handleClickOutside);
      return () => {
        document.removeEventListener("mousedown", handleClickOutside);
      };
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [edit])

  const handleUpdateStatus = (todoId: any, e: React.ChangeEvent<HTMLInputElement>) => {
    onChangeStatus(todoId, e)
  }
  const handleDeleteTodo = (todoId: any) => {
    onDeleteTodo(todoId)
  }
  const handleEditContent = () => {
    setEdit(true)
  }
  const onHandleTodoContent = (e: React.KeyboardEvent<HTMLSpanElement>) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      onUpdateTodoContent(todo.id, spanRef.current.textContent)
      setEdit(false)
    }
  }
  return (
    <div
      className={todo.status === TodoStatus.ACTIVE ? "ToDo__item" : "ToDo__item__completed"}
      onDoubleClick={handleEditContent}
    >
      <div className='Todo__content'>
        <input
          type="checkbox"
          checked={todo.status === TodoStatus.COMPLETED}
          onChange={e => handleUpdateStatus(todo.id, e)}
          onDoubleClick={e => e.stopPropagation()}
        />
        <span
          ref={spanRef}
          contentEditable={edit}
          onDoubleClick={handleEditContent}
          onKeyDown={onHandleTodoContent}
          suppressContentEditableWarning={true}
          children={todo.content}
        />
      </div>
      <button
        className="Todo__delete"
        onClick={() => handleDeleteTodo(todo.id)}
      >
        X
      </button>
    </div >
  );
};

export default TodoItem;