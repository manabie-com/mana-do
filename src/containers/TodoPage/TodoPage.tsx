import React, { useEffect, useReducer, useState } from "react";
import reducer, { initialState } from "../../store/reducer";
import { EnhanceTodoStatus } from "../../models/todo";
import TodoFilter from "../../components/TodoFilter/TodoFilter";
import TodoCreation from "../../components/TodoCreation/TodoCreation";
import TodoToolbar from "../../components/TodoToolbar/TodoToolbar";
import Service from "../../service";
import { setTodos } from "../../store/actions";
import TodoItem from "../../components/TodoItem/TodoItem";
import { ALL } from "../../common/constants";
import "./TodoPage.scss";

const TodoPage = () => {
  const [{ todos }, dispatch] = useReducer(reducer, initialState);
  const [showing, setShowing] = useState<EnhanceTodoStatus>(ALL);

  useEffect(() => {
    (async () => {
      const resp = await Service.getTodos();
      dispatch(setTodos(resp || []));
    })();
  }, []);

  const todoFilter = todos.filter(
    (todo) => showing === ALL || showing === todo.status
  );

  return (
    <div className="Todo__container">
      <TodoFilter todos={todos} showing={showing} setShowing={setShowing} />
      <TodoToolbar todos={todoFilter} dispatch={dispatch} />
      <div className="Todo__list">
        {todoFilter.map((todo, index) => {
          return <TodoItem key={index} todo={todo} dispatch={dispatch} />;
        })}
      </div>
      <TodoCreation todos={todoFilter} dispatch={dispatch} />
    </div>
  );
};

export default TodoPage;
