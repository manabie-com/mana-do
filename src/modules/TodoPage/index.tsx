import React, { useEffect, useReducer, useRef, useState } from 'react';
import "./styles.css"
import reducer, { initialState } from '../../store/reducer';
import { setTodos } from '../../store/actions';
import { ServiceFE } from '../../service';
import { TodoStatus, Todo } from '../../models/todo';
import TodoItem from './TodoItem';

type EnhanceTodoStatus = TodoStatus | 'ALL';

const ToDoPage = () => {
  const [{ todos }, dispatch] = useReducer(reducer, initialState);
  const [showing, setShowing] = useState<EnhanceTodoStatus>('ALL');
  const inputRef = useRef<any>(null);

  useEffect(() => {
    getTodos()
  }, [])

  const checkSelectAllTodo = (todos: Todo[]) => {
    for (const todo of todos) {
      if (todo.status === TodoStatus.ACTIVE) {
        return false
      }
    }
    return true
  }

  const getTodos = async () => {
    await ServiceFE.getTodos()
      .then(
        res => {
          dispatch(setTodos(res || []));
        }
      )
      .catch(error => console.log(error))
  }

  const onCreateTodo = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter' && inputRef.current.value) {
      await ServiceFE.createTodo(inputRef.current.value);
      getTodos()
      inputRef.current.value = "";
    }
  }

  const onUpdateTodoStatus = async (todoId: any, e: React.ChangeEvent<HTMLInputElement>) => {
    await ServiceFE.updateTodoStatus(todoId, e.target.checked)
    getTodos()
  }

  const onToggleAllTodo = async (e: React.ChangeEvent<HTMLInputElement>) => {
    await ServiceFE.toggleAllTodo(e.target.checked)
    getTodos()
  }

  const onDeleteAllTodo = async () => {
    await ServiceFE.deleteAllTodo()
    getTodos()
  }

  const onDeleteTodo = async (todoId: any) => {
    await ServiceFE.deleteTodo(todoId)
    getTodos()
  }
  const onUpdateTodoContent = async (todoId: any, content: any) => {
    await ServiceFE.updateTodoContent(todoId, content)
    getTodos()
  }
  return (
    <div className="ToDo__container">
      <h2>To do list</h2>
      <div className="Todo__creation">
        <input
          ref={inputRef}
          className="Todo__input"
          placeholder="What need to be done?"
          onKeyDown={onCreateTodo}
        />
      </div>
      <div className="Todo__toolbar">
        {todos.length > 0 ?
          <input
            checked={checkSelectAllTodo(todos)}
            type="checkbox"
            onChange={onToggleAllTodo}
          /> : <div />
        }
        <div className="Todo__tabs">
          <button
            className={showing === 'ALL' ? "Selected__btn" : "Action__btn"}
            onClick={() => setShowing('ALL')}
          >
            All
          </button>
          <button className={showing === TodoStatus.ACTIVE ? "Selected__btn" : "Action__btn"}
            onClick={() => setShowing(TodoStatus.ACTIVE)}
          >
            Active
          </button>
          <button className={showing === TodoStatus.COMPLETED ? "Selected__btn" : "Action__btn"}
            onClick={() => setShowing(TodoStatus.COMPLETED)}
          >
            Completed
          </button>
        </div>
        <button className='delete__all__btn' onClick={onDeleteAllTodo}>
          Clear all todos
        </button>
      </div>
      <div className="ToDo__list">
        {
          todos && todos
            .filter((todo) => showing === 'ALL' || showing === todo.status)
            .map((todo) => {
              return (
                <TodoItem
                  key={todo.id}
                  todo={todo}
                  onChangeStatus={onUpdateTodoStatus}
                  onDeleteTodo={onDeleteTodo}
                  onUpdateTodoContent={onUpdateTodoContent}
                />
              );
            })
        }
      </div>
    </div>
  );
};

export default ToDoPage;