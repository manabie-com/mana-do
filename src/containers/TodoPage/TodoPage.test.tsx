import React, { FC, ReactElement } from "react";
import { configure, mount } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import TodoPage from "./TodoPage";

configure({ adapter: new Adapter() });

describe("TodoPage", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(<TodoPage />);
  });

  afterEach(() => {
    wrapper.unmount();
    jest.clearAllMocks();
  });

  it("Should renders TodoPage correctly", () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
