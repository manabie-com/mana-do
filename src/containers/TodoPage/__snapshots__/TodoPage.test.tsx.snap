// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoPage Should renders TodoPage correctly 1`] = `
<div
  className="Todo__container"
>
  <div
    className="Todo__filter"
  >
    <div
      className="Todo__filter_item"
      id="filter-all"
      onClick={[Function]}
    >
      <h4>
        All
      </h4>
      <small>
        0
         Items out of 
        0
      </small>
      <div
        className="loading-bar"
      >
        <div
          className="loading-bar--progress"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <span
            className="first"
          />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span
            className="last"
          />
        </div>
      </div>
    </div>
    <div
      className="Todo__filter_item"
      id="filter-active"
      onClick={[Function]}
    >
      <h4>
        Active
      </h4>
      <small>
        0
         Items out of 
        0
      </small>
      <div
        className="loading-bar"
      >
        <div
          className="loading-bar--progress"
          style={
            Object {
              "width": "0%",
            }
          }
        >
          <span
            className="first"
          />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span
            className="last"
          />
        </div>
      </div>
    </div>
    <div
      className="Todo__filter_item"
      id="filter-complete"
      onClick={[Function]}
    >
      <h4>
        Compeleted
      </h4>
      <small>
        0
         Items out of 
        0
      </small>
      <div
        className="loading-bar"
      >
        <div
          className="loading-bar--progress"
          style={
            Object {
              "width": "0%",
            }
          }
        >
          <span
            className="first"
          />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
          <span
            className="last"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="Todo__toolbar"
  >
    <button
      className="Todo__toolbar_button Todo__toolbar_button--complete"
      onClick={[Function]}
    >
      Uncomplete All Items
    </button>
    <button
      className="Todo__toolbar_button Todo__toolbar_button--delete"
      onClick={[Function]}
    >
      Delete All Items
    </button>
  </div>
  <div
    className="Todo__list"
  />
  <div
    className="Todo__creation"
  >
    <input
      className="Todo__creation_input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Please enter your to do list"
      value=""
    />
  </div>
</div>
`;
